frontend/src/scenes/instance/AsyncMigrations/asyncMigrationsLogic.ts

```diff
@@ -43,21 +49,42 @@ export interface AsyncMigration {
     finished_at: string
     posthog_min_version: string
     posthog_max_version: string
+    error_cnt: number
 }
 
-export const asyncMigrationsLogic = kea<asyncMigrationsLogicType<AsyncMigration, AsyncMigrationsTab>>({
+export const asyncMigrationsLogic = kea<
+    asyncMigrationsLogicType<AsyncMigration, AsyncMigrationError, AsyncMigrationsTab>
+>({
     path: ['scenes', 'instance', 'AsyncMigrations', 'asyncMigrationsLogic'],
     actions: {
         triggerMigration: (migrationId: number) => ({ migrationId }),
+        resumeMigration: (migrationId: number) => ({ migrationId }),
+        rollbackMigration: (migrationId: number) => ({ migrationId }),
         forceStopMigration: (migrationId: number) => ({ migrationId }),
+        forceStopMigrationWithoutRollback: (migrationId: number) => ({ migrationId }),
         setActiveTab: (tab: AsyncMigrationsTab) => ({ tab }),
         updateSetting: (settingKey: string, newValue: string) => ({ settingKey, newValue }),
     },
 
     reducers: {
         activeTab: [AsyncMigrationsTab.Management, { setActiveTab: (_, { tab }) => tab }],
+        asyncMigrationErrorsLoading: [
+            {} as Record<number, boolean>,
+            {
+                loadAsyncMigrationErrors: (state, migrationId) => {
+                    return { ...state, [migrationId]: true }
+                },
+                loadAsyncMigrationErrorsSuccess: (state, payload) => {
+                    return { ...state, [payload.payload]: false }
+                },
+                loadAsyncMigrationErrorsFailure: (state, payload) => {
+                    console.log(payload)
```

Comment:
here's a problem - I don't have enough info here (i.e. I don't know what ID the loading function was called on)
<img width="717" alt="Screen Shot 2022-01-24 at 13 49 21" src="https://user-images.githubusercontent.com/890921/150786188-3f8a7184-6df2-40f6-b550-d07c6a4f8382.png">

Reply:
Hm, maybe the API's error response (which we use as `errorObject` verbatim) should include a `lookup` property, which would be the lookup value that was used (in this case, migration ID).

Reply:
Fixed this by manually defining listeners & everything instead of using loaders